src/panels/calendar/ha-big-calendar.js

```diff
@@ -0,0 +1,65 @@
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { PolymerElement } from '@polymer/polymer/polymer-element.js';
+
+/* eslint-disable */
+import { render } from 'react-dom';
+import React from 'react';
+/* eslint-enable */
+import BigCalendar from 'react-big-calendar';
+import moment from 'moment';
+
+import '../../resources/ha-style.js';
+
+BigCalendar.setLocalizer(BigCalendar.momentLocalizer(moment));
+
+class HABigCalendar extends PolymerElement {
+  static get template() {
+    return html`
+      <link rel="stylesheet" href="/static/panels/calendar/react-big-calendar.css">
+      <style>
+        div#root {
+          height: 100%;
+          width: 100%;
+        }
+      </style>
+      <div id="root"></div>`;
+  }
+
+  update(events) {
+    const allViews = BigCalendar.Views.values;
+
+    const BCElement = React.createElement(
+      BigCalendar,
+      { events: events,
+        views: allViews,
+        popup: true,
+        onNavigate: this.dateUpdated,
+        eventPropGetter: this.setEventStyle,
+        defaultDate: new Date(), }
```

Comment:
this is the current selected date?!
you reset this on every update

Reply:
It doesn't matter. defaultDate is only used when the component is initialized. After that it maintains it's own state and updates us by calling `onNavigate`